lib/jekyll/filters/url_filters.rb

```diff
@@ -10,16 +10,20 @@ module URLFilters
       # Returns the absolute URL as a String.
       def absolute_url(input)
         return if input.nil?
-
-        input = input.url if input.respond_to?(:url)
-        return input if Addressable::URI.parse(input.to_s).absolute?
-
-        site = @context.registers[:site]
-        return relative_url(input) if site.config["url"].nil?
-
-        Addressable::URI.parse(
-          site.config["url"].to_s + relative_url(input)
-        ).normalize.to_s
+        cache = (@context.registers[:cached_absolute_urls] = {})
```

Comment:
Wouldn't this line cause `@context.registers[:cached_absolute_urls]` to be reset to `{}` on every call to the filter..?
Or am I reading it wrong?

Reply:
Gah, good eyes! It's the `relative_url` caching that makes all the difference for the Cockroach docs, since `relative_url` is used a fair bit in the base layout. I added caching to `absolute_url` for good measure (presumably there's some other Jekyll site out there that's using absolute URLs instead of relative URLs everywhere), and just botched the job. Thanks! Should be fixed now.